src/parser/expression.js

```diff
@@ -896,6 +896,14 @@ pp.parseObj = function (isPattern, refShorthandDefaultPos) {
   node.properties = [];
   this.next();
 
+  // `for` keyword begins an object comprehension.
+  if (this.hasPlugin("lightscript") && this.match(tt._for)) {
+    // ...however, `{ for: x }` is a legal JS object.
```

Comment:
I've been looking at the lookahead code and I  think we can actually do this in a performant way. Since our lookahead here isn't actually relying on the whole parser state, nor do we call any functions during the lookahead, we may be able to do a "cheap" lookahead where we ignore the parser state altogether.